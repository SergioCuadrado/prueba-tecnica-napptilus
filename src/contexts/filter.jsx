import { createContext, useState } from 'react'

export const FilterContext = createContext()

export const ProductsFilterProvider = ({ children }) => {
  const [filter, setProductsFilter] = useState({
    search: ''
  })

  return (
    <FilterContext.Provider value={{ filter, setProductsFilter }}>
      {children}
    </FilterContext.Provider>
  )
}
