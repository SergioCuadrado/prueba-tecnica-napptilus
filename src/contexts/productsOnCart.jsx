import { createContext, useState } from 'react'

export const ProductsOnCartContext = createContext()

export const ProductsOnCartProvider = ({ children }) => {
  const [productsOnCart, setProductsOnCart] = useState(0)

  return (
    <ProductsOnCartContext.Provider value={{ productsOnCart, setProductsOnCart }}>
      {children}
    </ProductsOnCartContext.Provider>
  )
}
