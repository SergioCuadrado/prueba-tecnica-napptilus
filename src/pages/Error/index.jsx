import { Link, useRouteError } from 'react-router-dom'

import './styles.css'

export const ErrorPage = () => {
  const error = useRouteError()
  return (
    <section className='error-page'>
        <h2>{error?.status} Error</h2>
        <p>{error?.data}</p>
        <Link to='/'>Volver a la p√°gina inicial</Link>
    </section>
  )
}
