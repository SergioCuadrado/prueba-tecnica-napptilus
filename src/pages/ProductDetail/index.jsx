import { useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { DetailProduct } from '../../components/DetailProduct'
import { Loader } from '../../components/Loader'
import { useProduct } from '../../hooks/useProduct'

const DETAILS_IN_SPANISH = {
  brand: 'Marca',
  model: 'Modelo',
  price: 'Precio',
  cpu: 'CPU',
  ram: 'RAM',
  os: 'Sistema Operativo',
  screenResolution: 'Resolución de pantalla',
  battery: 'Batería',
  primaryCamera: 'Cámara principal',
  secondaryCamera: 'Cámara secundaria',
  dimensions: 'Dimensiones',
  weight: 'Peso'
}

export const ProductDetail = () => {
  const { id } = useParams()
  const { getDetailProduct, product, error, loading } = useProduct()

  useEffect(() => {
    getDetailProduct({ id })
  }, [])

  const descriptionProducts = product
    ? Object.entries(product?.description).map((desc) => {
      const name = DETAILS_IN_SPANISH[desc[0]]
      const value = desc[1]
      return { name, value }
    }).filter((desc) => {
      return desc.name !== undefined
    })
    : null

  const optionsProduct = product ? product.options : null

  if (error) return <DetailErrorProduct error={error} />
  if (loading) return <Loader />

  return (
    <>
      {product
        ? <DetailProduct product={product} descriptionProducts={descriptionProducts} optionsProduct={optionsProduct} errorValue={error} />
        : null
      }
    </>

  )
}

const DetailErrorProduct = ({ error }) => {
  return (
    <>
      <h2>{error}</h2>
      <p>Por favor, intenta nuevamente más tarde</p>
    </>
  )
}
