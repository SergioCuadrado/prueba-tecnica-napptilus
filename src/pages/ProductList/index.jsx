import { Filter } from '@/components/Filter'
import { Products } from '@/components/Products'
import { useFiltersProducts } from '@/hooks/useFiltersProducts'
import { useProducts } from '@/hooks/useProducts'
import { useEffect } from 'react'

export const ProductList = () => {
  const { filterProducts } = useFiltersProducts()
  const { getProducts, products, loading, error } = useProducts()
  let filteredProducts = []

  useEffect(() => {
    getProducts()
  }, [])

  if (!loading && products?.length > 0) filteredProducts = filterProducts(products)

  if (error) return <h2>{error?.message}</h2>

  return (
    <>
      <Filter />
      <Products products={filteredProducts} loading={loading} />
    </>
  )
}
