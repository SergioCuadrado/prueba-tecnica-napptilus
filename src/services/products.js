const { VITE_API_PRODUCTS } = import.meta.env
export const getListProducts = async () => {
  const { signal } = new AbortController()
  try {
    const response = await fetch(`${VITE_API_PRODUCTS}/product`, { signal })
    if (!response.ok) throw new Error('Error: No se pudo obtener los productos')
    if (signal.aborted) throw new Error('Petición cancelada, inténtelo de nuevo')

    const products = await response.json()

    return products.map((product) => ({
      id: product.id,
      brand: product.brand,
      model: product.model,
      price: product.price,
      image: product.imgUrl
    }))
  } catch (error) {
    if (error.name === 'AbortError') return
    throw new Error(error.message)
  }
}
