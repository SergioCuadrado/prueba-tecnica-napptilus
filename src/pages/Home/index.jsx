import { Outlet } from 'react-router-dom'
import { Header } from '@/components/Header'
import { ProductsOnCartProvider } from '../../contexts/productsOnCart'

export const Home = () => {
  return (
    <div className="page">
      <ProductsOnCartProvider>
        <Header />
        <Outlet />
      </ProductsOnCartProvider>
    </div>
  )
}
