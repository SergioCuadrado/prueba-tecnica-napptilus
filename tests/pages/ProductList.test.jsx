import { render } from '@testing-library/react'

import { ProductList } from '@/pages/ProductList'
import { FilterContext } from '@/contexts/filter'
import { ProductsContext } from '@/contexts/products'

describe('Pruebas en <ProductList />', () => {
  const filter = {
    search: ''
  }
  const setProductsFilterMock = jest.fn()
  const setProductsMock = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Snapshot de la pÃ¡gina', () => {
    const { container } = render(
    <FilterContext.Provider value={{ filter, setProductsFilter: setProductsFilterMock }}>
        <ProductsContext.Provider value={{ products: [], setProducts: setProductsMock }}>
            <ProductList />
        </ProductsContext.Provider>
    </FilterContext.Provider>

    )
    expect(container).toMatchSnapshot()
  })

  test('Cuando inicias por primera vez tiene que mostrarse el `skeleton` de react', () => {
    const { container } = render(
      <FilterContext.Provider value={{ filter, setProductsFilter: setProductsFilterMock }}>
        <ProductsContext.Provider value={{ products: [], setProducts: setProductsMock }}>
          <ProductList />
        </ProductsContext.Provider>
      </FilterContext.Provider>
    )

    expect(container.querySelector('.react-loading-skeleton')).toBeInTheDocument()
  })
})
