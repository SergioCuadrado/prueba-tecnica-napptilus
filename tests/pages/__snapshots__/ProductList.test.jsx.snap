// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductList /> Snapshot de la página 1`] = `
<div>
  <form
    class="form"
  >
    <label
      for=":r0:"
    >
      Búsqueda por marca / modelo:
    </label>
    <input
      id=":r0:"
      name=":r0:"
      placeholder="Liquid, Iconia..."
      type="text"
      value=""
    />
  </form>
  <ul
    class="products"
  >
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
    <li
      class="product"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton"
          style="height: 250px;"
        >
          ‌
        </span>
        <br />
      </span>
    </li>
  </ul>
</div>
`;
